### PR TITLE
Take the import/export feature out of tech preview

### LINKED ISSUE #1236: Revisit features in tech preview (import/export, image building)
https://docs.pulpproject.org/pulp_container/tech-preview.html

- Build an OCI image from a Containerfile
- Import/Export with synced container repositories

These features have been present in the plugin for a longer period. We should strive to mark them production ready without further ado.

### PR DESCRIPTION
**Pull request by lubosmj**
This is a continuous effort for
https://github.com/pulp/pulpcore/issues/3429.

closes #1236

### CODE REVIEW

**Review by ipanova — COMMENTED** [review]
That looks fine to me. However, as last 'pre-check', given that DavidN was looking into oci image builder feature, mind asking him in case he has found some major  red flags?
Another thing on my mind is - since RBAC roles refactor, we scope the artifact/content from repo permissions. Given that in OCI image building you need to provide path to the just uploaded artifact(s), it seems like this feature is limited to only admins? Or the user would need to make a content(presumably file content?) out of uploaded artifact and assign it to the repo( because users cannot simply upload content without providing destination to the repo). What i am trying to say that this workflow is not accessible to a regular ( non-admin) user https://docs.pulpproject.org/pulp_container/workflows/build-containerfile.html#build-an-oci-image

File: docs/index.rst [hunk 1]
```diff
@@ -29,14 +29,6 @@ Features
 * De-duplication of all saved content
 * Support disconnected and air-gapped environments with the Pulp Import/Export facility for container repositories
 
-Tech Preview
```
[the remark this answers is not in the document]
Comment (lubosmj) [reply docs/index.rst:32]: If there will be any features in tech preview, I will add the paragraph back. Right now, it only clutters the documentation because we do not have any features in tech preview.

File: docs/tech-preview.rst [hunk 2]
```diff
@@ -1,7 +0,0 @@
-Tech previews
-=============
```
[the remark this answers is not in the document]
Comment (ipanova) [reply docs/tech-preview.rst:2]: so people can check what the current list of feature of tech preview. otherwise they do not know whether we just forgot to add such section or we did not foget and simply have no feature in tech preview. explicit is better than implicit.
Also think of yourself in the future and ease potential manual steps you will need to take. Are you going to add this https://github.com/pulp/pulp_container/pull/1287#discussion_r1202615902 and https://github.com/pulp/pulp_container/pull/1287#discussion_r1202617087 every time you get a feature in tech preview? And similarly, are you going to remove these 2 every time when you take out features from tech preview? :)

File: docs/index.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (ipanova) [reply docs/index.rst:32]: see https://github.com/pulp/pulp_container/pull/1287#discussion_r1202671587
I don't consider cluttering docs with one 3 liner paragraph.

**Comment by lubosmj** [comment]
Okay, fair points. Let me add a warning to the builder feature about specific admin usage.

File: docs/index.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (daviddavis) [reply docs/index.rst:32]: What would even be better would be a line like "There are no features currently in tech preview" I think.

**Comment by mdellweg** [comment]
Do we want to change the user interface to image builder so that a user specifies a repository containing the Containerfile (either FileContent or some new type)? Then we should not take it out of tech preview.

How about this: A file RepositoryVersion represents a subdirectory on a file system. That is exactly what podman/buildah/... expect as a build context. Let it contain the Containerfile __and__ all the other assets...

**Comment by ipanova** [comment]
> Do we want to change the user interface to image builder so that a user specifies a repository containing the Containerfile (either FileContent or some new type)? Then we should not take it out of tech preview.
> 
> How about this: A file RepositoryVersion represents a subdirectory on a file system. That is exactly what podman/buildah/... expect as a build context. Let it contain the Containerfile **and** all the other assets...

Given that after RBAC roles refactor we have sort of 'downgraded' the usage of this feature I am tempted to say we should focus on make it usable by users, not just admins. So, yes, we need to provide to the users a way to specify the build context.
My suggestion would be  to take  out of tech preview just import/export and bring the image builder feature discussion on  the next integration meeting soliciting feedback from @newswangerd

**Comment by lubosmj** [comment]
The build feature has not been updated for quite a long time. I do not see any problem with removing it from the tech preview even if we decide to change its API later. The "later" may or may never come.

File: docs/workflows/build-containerfile.rst [hunk 3]
```diff
@@ -65,3 +65,9 @@ Build an OCI image
 
 .. literalinclude:: ../_scripts/build_containerfile.sh
    :language: bash
+
+
+.. warning::
+    Users lacking read access to the ``artifacts`` endpoint may encounter restricted functionality
```
Comment (ipanova) [inline docs/workflows/build-containerfile.rst:71]: did you just say in other words that this is restricted to admin-only users? :)

**Review by ipanova — APPROVED** [review]

File: docs/workflows/build-containerfile.rst (hunk 3 above)
Comment (lubosmj) [reply docs/workflows/build-containerfile.rst:71]: Yes. Is it understandable? I could reword it to contain the "non-staff users" term.
